### PR TITLE
PR-Catalog-of-things-project

### PR DESCRIPTION
**Pull request by mmsesay**
In this project, we worked on the following:

- Recorded a [video](https://drive.google.com/file/d/1xuZBndNdDYJW_bFgswAooEzKuoUv2Fhn/view?usp=sharing ) where we explained the program and some pieces of codes we are proud of making.
- Implement the correct git-flow
- All methods visible in the diagram were implemented:
     - add_item method
     - can_be_archived
     - move_to_archive etc...
- Add unit tests for all implemented methods.
- Created a database schema.sql file with tables that will be analogical to the structure of the classes in your app.
- Create the app UI and allow users inputs
- Create a main.rb file that will serve as your console app entry-point.
- The following options were included:
      - List all books
      - List all music albums
      - List of games
      - List all genres (e.g 'Comedy', 'Thriller')
      - List all labels (e.g. 'Gift', 'New')
      - List all authors (e.g. 'Stephen King')
      - Add a book
      - Add a music album
      - Add a game
- All data were be preserved by saving collections in .json files.

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def initialize(publisher, publish_date, cover_state)
+    super(publish_date)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def can_be_archived?
+    @archived || cover_state == 'bad'
+  end
```
Comment (mikethreels) [inline classes/book.rb:14]: - [ ] the `can_be_archived?` method should return true if parent's method  returns true OR if cover_state equals to "bad". in this case, you're checking the `@archived` instance variable, this should be `super` like in `Game` class and `Music_album` class

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Highlights:
- Great job on the professional video, all of you having your camera on and your audio in order makes it a joy to watch.
- The readme file looks awesome with the gif.
- great job on this project in general apart from a typo it looks great.
- 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ] [OPTIONAL] in might be nice to add all the information when listing a book, game or music album, so for example when listing a book display all the information entered when adding a book. including the title, color, author etc.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb (hunk 1 above)
Comment (DeliceLydia) [reply classes/book.rb]: Addressed this!

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### You Rock!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**